File: dhandho-jsclient/src/main/typings/app/command/command.component.ts
import { Component } from '@angular/core';
import { HttpClientModule, HttpClient, HttpResponse } from '@angular/common/http';
import { CommandResponse } from './command-response';
import { LoggerService } from '../service/logger.service';
import { Observable } from 'rxjs/Observable';

@Component( {
    selector: 'app-command',
    templateUrl: './command.component.html',
    styleUrls: ['./command.component.css']
} )
export class CommandComponent {

    public title: string;

    public command: string;

    public responseArray: Array<CommandResponse> = <any>( new Array<any>() );

    url = '/web/cmd/';

    http: HttpClient;

    log: LoggerService;

    public constructor( http: HttpClient, log: LoggerService ) {
        this.title = 'Command Component';
        if ( this.command === undefined ) { this.command = null; }
        if ( this.http === undefined ) { this.http = null; }
        if ( this.log === undefined ) { this.log = null; }
        this.http = http;
        this.log = log;
    }

    public onResponse( requestTime: number, command: string, json: any ) {
        const typeP: PropertyDescriptor = Object.getOwnPropertyDescriptor( json, 'type' );
        const type: string = <string>typeP.value;
        const names: Array<string> = Object.getOwnPropertyNames( json );
        const rt: CommandResponse = new CommandResponse( requestTime, command, json );
        this.responseArray.unshift( rt );
    }

    public onButtonClick() {
        // console.log( "on Button Click,command:" + this.command );
        const command: string = this.command;
        const requestTime: number = /* currentTimeMillis */Date.now();
        const ores = this.http.post( this.url, command );

        ores.toPromise().then(( json ) => {
            console.log( json );
            this.onResponse( requestTime, command, json );
            this.log.debug( 'post response:' + json );
            return null;
        } ).catch(( t: any ) => {
            this.log.error( 'catched exception' );
            this.log.error( t );
            return null;
        } );
    }

    public showChart(): boolean {
        return true;
    }

    public isNumber( json: Object ): boolean {
        const type: string = typeof json;
        this.log.debug( 'typeof,json:%s is: %s', json, type );
        return /* equals */( <any>( ( o1: any, o2: any ) => {
            if ( o1 && o1.equals ) {
                return o1.equals( o2 );
            } else {
                return o1 === o2;
            }
        } )( type, 'number' ) );
    }
}

File: dhandho-jsclient/src/main/java/app/dhojsw/ng/command/CommandComponent.java
package app.dhojsw.ng.command;

import static def.dhojsw.jsnative.Globals.js_isNumber;
import def.angular.common_http.HttpClient;
import def.angular.core.Component;
import def.js.Array;
import def.js.PropertyDescriptor;
import def.rxjs.rxjs.Observable;

@Component(//
		selector = "app-command", //
		templateUrl = "./command.component.html", //
		styleUrls = { "./command.component.css" }//
)

public class CommandComponent {
	public String title;
	public String command;
	public Array<CommandResponse> responseArray = new Array<>();
	String url = "/web/cmd/";
	HttpClient http;

	public CommandComponent(HttpClient http) {
		this.title = "Command Component";
		this.http = http;
	}

	public void onResponse(double requestTime, String command, Object json) {
		PropertyDescriptor typeP = def.js.Object.getOwnPropertyDescriptor(json, "type");
		String type = (String) typeP.value;

		Array<def.js.String> names = def.js.Object.getOwnPropertyNames(json);
		CommandResponse rt = new CommandResponse(requestTime, command, json);
		this.responseArray.unshift(rt);
	}

	public void onButtonClick() {
		// console.log( "on Button Click,command:" + this.command );
		String command = this.command;
		double requestTime = def.js.Date.now();
		Observable<Object> ores = this.http.post(this.url, command);

		ores.toPromise().then((json) -> {
			// console.log( json );
			this.onResponse(requestTime, command, json);
			// this.log.debug( "post response:" + json );
			return null;
		})._catch((Object t) -> {
			// this.log.error( "catched exception" );
			// this.log.error( t );
			return null;
		});
	}

	public boolean showChart() {
		return true;
	}

	public boolean isNumber(Object json) {
		return js_isNumber(json);
	}
}
